backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/util/git/GitUtils.java

```
@@ -84,6 +98,21 @@ public static List<URI> getVSCChangedFiles(@Nullable Path baseDir) {
     }
   }
 
+  public static SonarLintBlameResult getBlameResult(Path projectBaseDir, Set<Path> projectBaseRelativeFilePaths, @Nullable UnaryOperator<String> fileContentProvider) {
+    return getBlameResult(projectBaseDir, projectBaseRelativeFilePaths, fileContentProvider, GitUtils::checkIfEnabled);
+  }
+
+  public static SonarLintBlameResult getBlameResult(Path projectBaseDir, Set<Path> projectBaseRelativeFilePaths, @Nullable UnaryOperator<String> fileContentProvider,
+    Predicate<Path> isEnabled) {
+    if (isEnabled.test(projectBaseDir)) {
+      LOG.debug("Using native git blame");
+      return blameFromNativeCommand(projectBaseDir, projectBaseRelativeFilePaths);
```

If the Git command throws an error instead of failing we should then fall back to JGit. This is currently not the case here (anymore).

---

Yes, but normally it should not because we test if git is installed and correct version with `checkIfEnabled`. We run native git and parse version output. It looks like an adequate check. So we assume blame command will also work.

One case when it won't - if there's no git repo, but in this case falling back to jgit also won't help.

---

This is a fair point! I was more concerned that `git blame` itself fails maybe due to incorrect inputs or so.

---

But it's we who provide the inputs. We fully in control. I don't see how it can be incorrect in normal scenario.